### DISCUSSION TITLE
Track0 signal absent after seek to cylinder 0

### DISCUSSION THREAD

**Discussion by mkatzNU** [keirf/greaseweazle#350] [Q&A]
I am using 2 SA-800-2 8" single sided drives.

Sometimes when I go to do a an operation on the drive I get the Track0 signal absent message.  If I power cycle the drives the problem goes away.

Does anyone have any ideas why this is happening?

**Comment by keirf** [comment]
You probably need to step slower. Try `gw delays --step 20000`

**Comment by mkatzNU** [comment]
The stepping speed of the SA800-2 drive is 8mS.  I tried setting it to 20mS and I am still seeing the error.

**Reply by keirf** [reply]
Sticky stepper? Have you tried increasing the settle time?

**Comment by aurum79argentum47** [comment]
I get the same thing (first times a charm, then fails until the drive is power cycled) on a correctly configured 40 track 5.25" Shugart SA400L which passes full disk diagnostics on my TI-99/4a Disk Manager 3. This is a SS SD drive with a DS DD diskette. This drive has had head cleaned, been lubricated, speed calibrated, has had photodiode and sensor replaced, has a new belt and has had microswitches replaced as PM.   It works flawlessly on the TI.

The suggested step and settling time changes do not work for me.   I have never been able to get GW to work reading or writing this drive (disks are empty when cataloged) or it companion a 35 track Shugart SA400 also in working order/

**Reply by keirf** [reply]
Does `gw reset` help when this happens?

**Reply by aurum79argentum47** [reply]
No, only power cycling.  Even then I get inconsistent behavior, and never a successul write or read. 

I should add that I have just discovered that in trying to run the "gw clean" routine as a test, on the first attempt it worked, second only through track 5 then the "Track0 signal absent after seek to cylinder 0" and 3rd time no head movement, must "Track0 signal absent ...".   I had high hopes for GW, but maybe these old Shugart drives are just too sloppy, despite working flawlessly in the TI-99/4A.

I would love to send you one to evaluate Keirf, sir, but I'm sure you have better things to do.

**Reply by keirf** [reply]
Well there are other 48tpi drives out there, so I suppose not working on one of the true originals isn't the end of the world. Gw must be violating some drive timing I suppose. The datasheet isn't really comprehensive on timing. Track to track seems to be 20ms. Perhaps another 20ms for settle. Step to track 0 assert doesn't seem to be specified.

When you say it doesn't read, does that mean if you took a raw dump of a track it would be empty of flux? That would be a worse interface problem. Like no drive terminations, or too strong drive terminations, for example.

We could probably build up from what does work, if anything, and get this working by tweaking or parameterising some delays in drive sequences in the gw firmware.

**Comment by aurum79argentum47** [comment]
Thank you kindly for your help.

When I do get tracks to read, the flux numbers are all identical and very
low, something like 200. I assume it's just background current and no
flux.  Newest 4.1 GW, latest firmware.

I'm using the IC terminating resistor with a single drive at the end of the
straight cable. Multiplex config switch is ON (therefore multiplex off if I
read the service manual correctly).  My Shugarts work well on the same
cable with the same terminating resistor on the ti.

 I just found another TI  peripheral test program that writes and reads
random sectors on the disc which runs correctly for both drives.
As I noted, I get the same GW behavior with the SA400 (35/10) and SA400L
(40/10).  Working with a new box of 3M DS/DD disc.  Yes I would love to get
this working and will follow any guidance you have for me.

Is Arlo Guthrie said, I ain't proud or tired.

**Reply by keirf** [reply]
What model of GW are you using? Can you record a raw track from a known-formatted disk and upload it for inspection?

**Comment by aurum79argentum47** [comment]
4.1 newest

Will do.

**Comment by aurum79argentum47** [comment]
4.1

Please find attached:

gw info command line output
gw read raw command line output (succeeded?)
a zip file of raw tracks
gw read ibm.scan command line output (failed)

I am unable to write with any options or seek to a cylinder after 1)
rebooting the PC 2) power cycling the drive 3) resetting GW
as shown here.   I consistently get a "No Index" or "Track 0 not
found" error.

test.txt

Thankyou,

Greg

On Sun, Apr 21, 2024 at 6:13 AM Keir Fraser ***@***.***>
wrote:

> What model of GW are you using? Can you record a raw track from a
> known-formatted disk and upload it for inspection?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/keirf/greaseweazle/discussions/350#discussioncomment-9178682>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AO66VDDU2OR2YDS76J27HZDY6ONPJAVCNFSM6AAAAAA3UVG2Q2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TCNZYGY4DE>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

C:\greaseweazle>gw info
Host Tools: 1.16.3
Device:
  Port:     COM5
  Model:    Greaseweazle V4.1
  MCU:      AT32F403A, 216MHz, 224kB SRAM
  Firmware: 1.5
  Serial:   GWB01431335976C010077C9705
  USB:      Full Speed (12 Mbit/s), 128kB Buffer
C:\greaseweazle>gw.exe read --diskdefs "C:\greaseweazle\diskdefs.cfg" --format=ibm.scan --adjust-speed=300rpm --fake-index=300 --tracks=c=0-34,35-40:h=0 "C:\greaseweazle\mydisk.hfe::bitrate=250"
Reading c=0-40:h=0 revs=2
Format ibm.scan
T0.0: IBM Empty from Raw Flux (10180 flux in 400.17ms)
T1.0: IBM Empty from Raw Flux (9106 flux in 400.26ms)
** FATAL ERROR:
Track0 signal asserted after seek to cylinder 2


C:\greaseweazle>gw.exe write --diskdefs "C:\greaseweazle\diskdefs.cfg" --format=ibm.360 --drive=A --tracks=step=2:c=0-39:h=0 "C:\Users\Greg\Downloads\BESTofXB.hfe"
Writing c=0-39:h=0:step=2
Format ibm.360
Command Failed: GetFluxStatus: No Index

C:\greaseweazle>gw seek 20
Command Failed: Seek: Track 0 not found
C:\greaseweazle>gw.exe read --diskdefs "C:\greaseweazle\diskdefs.cfg" --format=raw.125 --adjust-speed=300rpm --fake-index=300 --tracks=c=0-34,35-40:h=0 "C:\greaseweazle\mydisk01.1.raw"
Reading c=0-40:h=0 revs=1
Format raw.125
T0.0: Raw Bitcell (55460 bits, 200.00ms) from Raw Flux (2824 flux in 198.98ms)
T1.0: Raw Bitcell (55535 bits, 200.00ms) from Raw Flux (2937 flux in 200.43ms)
T2.0: Raw Bitcell (55454 bits, 200.00ms) from Raw Flux (2989 flux in 199.61ms)
T3.0: Raw Bitcell (55607 bits, 200.00ms) from Raw Flux (3642 flux in 199.54ms)
T4.0: Raw Bitcell (55584 bits, 200.00ms) from Raw Flux (3476 flux in 200.40ms)
T5.0: Raw Bitcell (55584 bits, 200.00ms) from Raw Flux (3293 flux in 198.05ms)
T6.0: Raw Bitcell (55595 bits, 200.00ms) from Raw Flux (3355 flux in 200.35ms)
T7.0: Raw Bitcell (55557 bits, 200.00ms) from Raw Flux (3586 flux in 200.44ms)
T8.0: Raw Bitcell (55557 bits, 200.00ms) from Raw Flux (3485 flux in 199.82ms)
T9.0: Raw Bitcell (55617 bits, 200.00ms) from Raw Flux (3677 flux in 200.46ms)
T10.0: Raw Bitcell (55584 bits, 200.00ms) from Raw Flux (3682 flux in 199.68ms)
T11.0: Raw Bitcell (55707 bits, 200.00ms) from Raw Flux (4373 flux in 200.50ms)
T12.0: Raw Bitcell (55690 bits, 200.00ms) from Raw Flux (4706 flux in 199.86ms)
T13.0: Raw Bitcell (55621 bits, 200.00ms) from Raw Flux (4091 flux in 200.50ms)
T14.0: Raw Bitcell (55526 bits, 200.00ms) from Raw Flux (4112 flux in 200.32ms)
T15.0: Raw Bitcell (55647 bits, 200.00ms) from Raw Flux (4184 flux in 200.47ms)
T16.0: Raw Bitcell (55585 bits, 200.00ms) from Raw Flux (4305 flux in 200.30ms)
T17.0: Raw Bitcell (55698 bits, 200.00ms) from Raw Flux (4404 flux in 200.48ms)
T18.0: Raw Bitcell (55603 bits, 200.00ms) from Raw Flux (4457 flux in 200.46ms)
T19.0: Raw Bitcell (55654 bits, 200.00ms) from Raw Flux (4513 flux in 200.45ms)
T20.0: Raw Bitcell (55627 bits, 200.00ms) from Raw Flux (4779 flux in 200.44ms)
T21.0: Raw Bitcell (55567 bits, 200.00ms) from Raw Flux (4750 flux in 200.46ms)
T22.0: Raw Bitcell (55628 bits, 200.00ms) from Raw Flux (4830 flux in 200.43ms)
T23.0: Raw Bitcell (55609 bits, 200.00ms) from Raw Flux (4909 flux in 200.42ms)
T24.0: Raw Bitcell (55603 bits, 200.00ms) from Raw Flux (5143 flux in 200.49ms)
T25.0: Raw Bitcell (55524 bits, 200.00ms) from Raw Flux (5295 flux in 199.97ms)
T26.0: Raw Bitcell (55535 bits, 200.00ms) from Raw Flux (5366 flux in 200.34ms)
T27.0: Raw Bitcell (55520 bits, 200.00ms) from Raw Flux (5471 flux in 200.30ms)
T28.0: Raw Bitcell (55415 bits, 200.00ms) from Raw Flux (5703 flux in 200.49ms)
T29.0: Raw Bitcell (55436 bits, 200.00ms) from Raw Flux (5725 flux in 200.46ms)
T30.0: Raw Bitcell (55400 bits, 200.00ms) from Raw Flux (5987 flux in 200.38ms)
T31.0: Raw Bitcell (55365 bits, 200.00ms) from Raw Flux (6258 flux in 200.49ms)
T32.0: Raw Bitcell (55365 bits, 200.00ms) from Raw Flux (6337 flux in 200.48ms)
T33.0: Raw Bitcell (55319 bits, 200.00ms) from Raw Flux (6311 flux in 200.49ms)
T34.0: Raw Bitcell (55261 bits, 200.00ms) from Raw Flux (6716 flux in 199.53ms)
T35.0: Raw Bitcell (55288 bits, 200.00ms) from Raw Flux (6738 flux in 200.50ms)
T36.0: Raw Bitcell (55102 bits, 200.00ms) from Raw Flux (6730 flux in 200.39ms)
T37.0: Raw Bitcell (55001 bits, 200.00ms) from Raw Flux (7071 flux in 200.17ms)
T38.0: Raw Bitcell (55111 bits, 200.00ms) from Raw Flux (7234 flux in 200.44ms)
T39.0: Raw Bitcell (55110 bits, 200.00ms) from Raw Flux (7263 flux in 199.78ms)
T40.0: Raw Bitcell (55097 bits, 200.00ms) from Raw Flux (7302 flux in 200.43ms)

**Comment by aurum79argentum47** [comment]
I should add that the disk read for raw tracks I sent was a working,
auto-booting TI-99/4A disk manager 3.0 DS/DD diskette written as SS/SD
running under the Extended Basic rom.

- Greg

**Comment by aurum79argentum47** [comment]
Wow.   Just figured out (with you help) that --step=50000 allows the disk
clean function to complete without the "Track zero not found" error!
 Please check this out.

C:\greaseweazle>gw delays --select=20
Select Delay: 20us
Step Delay: 10000us
Settle Time: 15ms
Motor Delay: 750ms
Watchdog: 10000ms

C:\greaseweazle>gw clean --cyls=40
Pass 0: 4 Command Failed: Seek: Track 0 not found

C:\greaseweazle>gw reset

C:\greaseweazle>gw delays
Select Delay: 10us
Step Delay: 10000us
Settle Time: 15ms
Motor Delay: 750ms
Watchdog: 10000ms

C:\greaseweazle>gw delays --step=50000
Select Delay: 10us
Step Delay: 50000us
Settle Time: 15ms
Motor Delay: 750ms
Watchdog: 10000ms

C:\greaseweazle>gw clean --cyls=40
Pass 0: 4 0 9 5 14 10 19 15 24 20 29 25 34 30 39 35
Pass 1: 4 0 9 5 14 10 19 15 24 20 29 25 34 30 39 35
Pass 2: 4 0 9 5 14 10 19 15 24 20 29 25 34 30 39 3

**Reply by keirf** [reply]
That's great. Possibly you can shrink the step time some, and possibly stretch the settle time a bit. But there's probably some time taken for the track 0 sensor to actuate as I believe it's a mechanical switch. Hence that probably extends the required step delay. Especially because this drive will not stop stepping at track 0 (except via a mechanical stop). The stepper will happily switch phase, and this also turns off the track 0 signal. Which would cause Greaseweazle to get very confused I'm sure.

**Reply by keirf** [reply]
By the way if you still want me to look at a flux file then you will need to attach it via a message sent via the GitHub web site. Email attachments get stripped.

**Comment by aurum79argentum47** [comment]
Ok, right away.

**Reply by aurum79argentum47** [reply]
Please find attached:

gw info command line output
gw read raw command line output (succeeded?)
a zip file of raw tracks
gw read ibm.scan command line output (failed)

I am unable to write with any options or seek to a cylinder after 1) rebooting the PC 2) power cycling the drive 3) resetting GW
as shown here.   I consistently get a "No Index" or "Track 0 not found" error.

test.txt

Thank you,

Greg

[greaseweazle gw info.txt](https://github.com/keirf/greaseweazle/files/15052655/greaseweazle.gw.info.txt)
[greaseweazle gw read ibmdotscan.txt](https://github.com/keirf/greaseweazle/files/15052656/greaseweazle.gw.read.ibmdotscan.txt)
[greaseweazle gw read raw.txt](https://github.com/keirf/greaseweazle/files/15052657/greaseweazle.gw.read.raw.txt)
[mydiskraw.zip](https://github.com/keirf/greaseweazle/files/15052659/mydiskraw.zip)
[test.txt](https://github.com/keirf/greaseweazle/files/15052660/test.txt)

**Comment by aurum79argentum47** [comment]
Changing the step and settle delays as you recommended, I am able to write the drive as shown below.   However, the flux currents look low and the disk is apparently empty when catalogued with Disk Manager 3.0 on the TI-99/4A.   What might be going on?

C:\greaseweazle>gw delays --step=40000
Select Delay: 10us
Step Delay: 40000us
Settle Time: 15ms
Motor Delay: 750ms
Watchdog: 10000ms

C:\greaseweazle>gw delays --settle=40
Select Delay: 10us
Step Delay: 40000us
Settle Time: 40ms
Motor Delay: 750ms
Watchdog: 10000ms

C:\greaseweazle>gw.exe write --diskdefs "C:\greaseweazle\diskdefs.cfg" --format=raw.125 --drive=A --fake-index=300 --tracks=step=2:c=0-39:h=0 "C:\Users\Greg\Downloads\BESTofXB.hfe"
Writing c=0-39:h=0:step=2
Format raw.125
T0.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T1.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T2.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T3.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T4.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T5.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T6.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T7.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T8.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T9.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T10.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T11.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T12.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T13.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T14.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T15.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T16.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T17.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T18.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T19.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T20.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T21.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T22.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T23.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T24.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T25.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T26.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T27.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T28.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T29.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T30.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T31.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T32.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T33.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T34.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T35.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T36.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T37.0: Writing Track (Flux: 200.0ms period, 220.0 ms total, Write all)
T38.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
T39.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Write all)
No tracks verified (Reason: Verify unavailable)

**Reply by keirf** [reply]
You might want to get read working before doing write. Do you have a known good disk to dump?

**Reply by aurum79argentum47** [reply]
Sorry - I've never been able to dump a disk using GW.     I will see if I have a TI utility which will do this on my FinalGrom99 tools.

Still struggling with:

TI-side: CFHDXS1
PC-side: HDX Server

which would allow me to do this with VDM99 on the PC.

**Reply by aurum79argentum47** [reply]
Oh, I understand.  Yes, the raw files I sent you in the .zip are from a known good diskette copy of TI Disk Manager 3.0

**Comment by aurum79argentum47** [comment]
Ok, here is the latest (successful) attempt to dump the working DM 3.0 diskette.   Step could not be too small or too large in order for this to work.   27000 seems optimal.
[mydisk_hfe.zip](https://github.com/keirf/greaseweazle/files/15052896/mydisk_hfe.zip)

Select Delay: 10us
Step Delay: 27000us
Settle Time: 15ms
Motor Delay: 750ms
Watchdog: 10000ms

C:\greaseweazle>gw.exe read --diskdefs "C:\greaseweazle\diskdefs.cfg" --format=ibm.scan --adjust-speed=300rpm --fake-index=300 --tracks=c=0-34,35-40:h=0 "C:\greaseweazle\mydisk.hfe::bitrate=250"
Reading c=0-40:h=0 revs=2
Format ibm.scan
T0.0: IBM Empty from Raw Flux (330 flux in 398.02ms)
T1.0: IBM Empty from Raw Flux (350 flux in 398.95ms)
T2.0: IBM Empty from Raw Flux (341 flux in 398.70ms)
T3.0: IBM Empty from Raw Flux (357 flux in 396.14ms)
T4.0: IBM Empty from Raw Flux (419 flux in 399.86ms)
T5.0: IBM Empty from Raw Flux (333 flux in 397.62ms)
T6.0: IBM Empty from Raw Flux (441 flux in 398.96ms)
T7.0: IBM Empty from Raw Flux (415 flux in 393.29ms)
T8.0: IBM Empty from Raw Flux (399 flux in 398.98ms)
T9.0: IBM Empty from Raw Flux (479 flux in 393.10ms)
T10.0: IBM Empty from Raw Flux (455 flux in 399.58ms)
T11.0: IBM Empty from Raw Flux (469 flux in 399.46ms)
T12.0: IBM Empty from Raw Flux (489 flux in 400.37ms)
T13.0: IBM Empty from Raw Flux (488 flux in 399.14ms)
T14.0: IBM Empty from Raw Flux (539 flux in 397.65ms)
T15.0: IBM Empty from Raw Flux (544 flux in 398.99ms)
T16.0: IBM Empty from Raw Flux (588 flux in 393.50ms)
T17.0: IBM Empty from Raw Flux (607 flux in 399.03ms)
T18.0: IBM Empty from Raw Flux (584 flux in 399.95ms)
T19.0: IBM Empty from Raw Flux (623 flux in 393.47ms)
T20.0: IBM Empty from Raw Flux (677 flux in 400.32ms)
T21.0: IBM Empty from Raw Flux (690 flux in 398.02ms)
T22.0: IBM Empty from Raw Flux (677 flux in 397.03ms)
T23.0: IBM Empty from Raw Flux (782 flux in 399.86ms)
T24.0: IBM Empty from Raw Flux (810 flux in 395.44ms)
T25.0: IBM Empty from Raw Flux (731 flux in 400.31ms)
T26.0: IBM Empty from Raw Flux (862 flux in 400.43ms)
T27.0: IBM Empty from Raw Flux (792 flux in 400.08ms)
T28.0: IBM Empty from Raw Flux (886 flux in 399.50ms)
T29.0: IBM Empty from Raw Flux (888 flux in 400.47ms)
T30.0: IBM Empty from Raw Flux (941 flux in 396.43ms)
T31.0: IBM Empty from Raw Flux (940 flux in 399.61ms)
T32.0: IBM Empty from Raw Flux (1009 flux in 400.43ms)
T33.0: IBM Empty from Raw Flux (1102 flux in 399.25ms)
T34.0: IBM Empty from Raw Flux (1261 flux in 400.38ms)
T35.0: IBM Empty from Raw Flux (1162 flux in 400.50ms)
T36.0: IBM Empty from Raw Flux (1155 flux in 397.49ms)
T37.0: IBM Empty from Raw Flux (1136 flux in 398.42ms)
T38.0: IBM Empty from Raw Flux (1217 flux in 398.19ms)
T39.0: IBM Empty from Raw Flux (1250 flux in 398.01ms)
T40.0: IBM Empty from Raw Flux (1218 flux in 399.01ms)

**Reply by keirf** [reply]
Try `gw read --tracks=c=0-39:h=0 name.scp` this will get a nice raw dump to a flux image file. A few hundred flux per track is far too low though. So I guess something is up.

**Reply by keirf** [reply]
Are you aware of drive select options by the way, for example `--drive=0`? It's not certain the default of drive a will be correct here. Is the disk definitely spinning?

**Reply by aurum79argentum47** [reply]
Will try.

Yes, I have the correct drive selected and it is actively reading, head stepping and spinning.

**Reply by aurum79argentum47** [reply]
.scp results for GW command below attached.  As you say, flux still looks low.

[dm3_scp.zip](https://github.com/keirf/greaseweazle/files/15052939/dm3_scp.zip)


C:\greaseweazle>gw delays
Select Delay: 10us
Step Delay: 27000us
Settle Time: 15ms
Motor Delay: 750ms
Watchdog: 10000ms

C:\greaseweazle>gw.exe read --diskdefs "C:\greaseweazle\diskdefs.cfg" --adjust-speed=300rpm --fake-index=300 --tracks=c=0-34,35-40:h=0 "C:\greaseweazle\dm3.scp"
Reading c=0-40:h=0 revs=3
T0.0: Raw Flux (479 flux in 600.24ms)
T1.0: Raw Flux (555 flux in 599.36ms)
T2.0: Raw Flux (567 flux in 592.88ms)
T3.0: Raw Flux (499 flux in 600.01ms)
T4.0: Raw Flux (520 flux in 598.88ms)
T5.0: Raw Flux (524 flux in 597.66ms)
T6.0: Raw Flux (626 flux in 600.21ms)
T7.0: Raw Flux (574 flux in 598.36ms)
T8.0: Raw Flux (602 flux in 599.57ms)
T9.0: Raw Flux (733 flux in 595.65ms)
T10.0: Raw Flux (870 flux in 600.68ms)
T11.0: Raw Flux (816 flux in 599.82ms)
T12.0: Raw Flux (694 flux in 594.96ms)
T13.0: Raw Flux (787 flux in 600.70ms)
T14.0: Raw Flux (827 flux in 600.82ms)
T15.0: Raw Flux (827 flux in 600.19ms)
T16.0: Raw Flux (900 flux in 600.91ms)
T17.0: Raw Flux (880 flux in 600.89ms)
T18.0: Raw Flux (1071 flux in 597.30ms)
T19.0: Raw Flux (997 flux in 599.26ms)
T20.0: Raw Flux (929 flux in 600.57ms)
T21.0: Raw Flux (1186 flux in 600.42ms)
T22.0: Raw Flux (1205 flux in 598.82ms)
T23.0: Raw Flux (1165 flux in 600.06ms)
T24.0: Raw Flux (1049 flux in 599.55ms)
T25.0: Raw Flux (1339 flux in 597.65ms)
T26.0: Raw Flux (1297 flux in 600.42ms)
T27.0: Raw Flux (1333 flux in 598.54ms)
T28.0: Raw Flux (1284 flux in 595.82ms)
T29.0: Raw Flux (1316 flux in 600.89ms)
T30.0: Raw Flux (1386 flux in 598.99ms)
T31.0: Raw Flux (1341 flux in 595.88ms)
T32.0: Raw Flux (1424 flux in 600.91ms)
T33.0: Raw Flux (1531 flux in 598.47ms)
T34.0: Raw Flux (1546 flux in 600.79ms)
T35.0: Raw Flux (1625 flux in 600.97ms)
T36.0: Raw Flux (1822 flux in 598.55ms)
T37.0: Raw Flux (1977 flux in 597.11ms)
T38.0: Raw Flux (1782 flux in 600.81ms)
T39.0: Raw Flux (1794 flux in 600.37ms)
T40.0: Raw Flux (1740 flux in 600.76ms)

**Reply by aurum79argentum47** [reply]
Weird (to me) that flux increases with each subsequent cylinder.

**Reply by keirf** [reply]
Yeah that's just noise really, albeit with a lot of samples around 2 microseconds. 

Why the --adjust-speed and especially the --fake-index options? Can you get a raw dump without those? The drive really should be generating index pulses. If it isn't, again, there's a broader problem.

**Reply by aurum79argentum47** [reply]
Works without --adjust -speed, but I get "NoIindex" when I drop the --fake-index option.

C:\greaseweazle>gw.exe read --diskdefs "C:\greaseweazle\diskdefs.cfg" --tracks=c=0-34,35-40:h=0 "C:\greaseweazle\dm3.scp"
Reading c=0-40:h=0 revs=3
Command Failed: GetFluxStatus: No Index

**Reply by aurum79argentum47** [reply]
As I noted, I can write, save and read Basic program files and data files, copy disks, and successfully run TI diagnostics on both Shugart 400 and 400L diskette drives.    One (the 400) has been refurbished to the best of my ability.  I really appreciate your patience and help with this!

**Reply by keirf** [reply]
Since these drives do have an index sensor, you need to work out why no index pulses are being received. Wrong select line, wrong motor line, incorrect power to drive, electrical incompatibility, and etc etc 

It's pointless to go looking at read data if even the simple index signal line ain't working.

**Reply by aurum79argentum47** [reply]
Ok, I'm out of my depth here.  I will do more research on these issuesand try to educate myself.  Thanks again.

Greg George
Sr. Research Chemist
WInner, 2020 International R&D 100 Award. Process & Prototyping

(but know enough about electronics to just to be dangerous :)

**Comment by aurum79argentum47** [comment]
Ok

Similar issues on a Mitsubishi MF504C-368U, but I have ignored these
because I'm using a ds/dd disc in this high density drive.

Wonder if my greaseweasle is working.    I clearly need a less fussy drive
that is known to work to test this.

Thanks

**Reply by keirf** [reply]
Yes that Mitsubishi should work. Either as drive B or as drive A depending on whether your ribbon has a twist or not. Yes you need a baseline working drive. A more modern 5.25 drive or even 3.5 drive, then at least you'd know the Greaseweazle is working.

**Reply by keirf** [reply]
And it is fine to read SD/DD disks in a HD 5.25 drive, but you might need to specify --densel on teh gw read command line to send the correct density select signal on interface pin 2. However none of that matters for getting an index signal out of the HD drive -- density mismatch wouldn't matter at all for that.

**Comment by aurum79argentum47** [comment]
I understand that - thx.  I'm going to do more experimenting with the
Mitsubishi.

**Comment by aurum79argentum47** [comment]
Got it.   I'm going to keep after this. Thanks.

**Comment by aurum79argentum47** [comment]
So it appears that I get higher fluxes from the 1.2 Mitsubishi 504C-368U with a DS/DD diskette.   If you please, is this attached dump valid?   However, I am unable to write this back to a blank disk.

C:\greaseweazle>gw.exe read --diskdefs "C:\greaseweazle\diskdefs.cfg" --drive=0 --dd L --tracks=c=0-34,35-40:h=0 "C:\greaseweazle\dm3.scp"
Reading c=0-40:h=0 revs=3
T0.0: Raw Flux (112907 flux in 685.32ms)
T1.0: Raw Flux (150541 flux in 741.80ms)
T2.0: Raw Flux (148540 flux in 732.42ms)
T3.0: Raw Flux (148620 flux in 732.83ms)
T4.0: Raw Flux (148672 flux in 733.09ms)
T5.0: Raw Flux (143832 flux in 732.90ms)
T6.0: Raw Flux (144066 flux in 733.84ms)
T7.0: Raw Flux (130092 flux in 734.23ms)
T8.0: Raw Flux (130749 flux in 737.65ms)
T9.0: Raw Flux (134312 flux in 737.09ms)
T10.0: Raw Flux (134224 flux in 736.66ms)
T11.0: Raw Flux (138179 flux in 736.78ms)
T12.0: Raw Flux (138042 flux in 735.96ms)
T13.0: Raw Flux (134153 flux in 735.52ms)
T14.0: Raw Flux (134284 flux in 736.31ms)
T15.0: Raw Flux (134041 flux in 737.02ms)
T16.0: Raw Flux (133971 flux in 736.62ms)
T17.0: Raw Flux (133813 flux in 737.21ms)
T18.0: Raw Flux (133825 flux in 737.30ms)
T19.0: Raw Flux (128086 flux in 736.34ms)
T20.0: Raw Flux (126999 flux in 731.19ms)
T21.0: Raw Flux (138996 flux in 738.34ms)
T22.0: Raw Flux (138421 flux in 735.50ms)
T23.0: Raw Flux (149392 flux in 736.19ms)
T24.0: Raw Flux (148561 flux in 732.69ms)
T25.0: Raw Flux (148187 flux in 730.50ms)
T26.0: Raw Flux (148718 flux in 733.26ms)
T27.0: Raw Flux (148398 flux in 731.87ms)
T28.0: Raw Flux (148559 flux in 732.71ms)
T29.0: Raw Flux (148671 flux in 732.75ms)
T30.0: Raw Flux (148730 flux in 733.01ms)
T31.0: Raw Flux (148652 flux in 733.07ms)
T32.0: Raw Flux (148650 flux in 733.06ms)
T33.0: Raw Flux (148672 flux in 733.13ms)
T34.0: Raw Flux (148504 flux in 732.52ms)
T35.0: Raw Flux (148768 flux in 732.99ms)
T36.0: Raw Flux (148644 flux in 732.53ms)
T37.0: Raw Flux (148573 flux in 731.54ms)
T38.0: Raw Flux (148799 flux in 732.84ms)
T39.0: Raw Flux (148454 flux in 731.96ms)
T40.0: Raw Flux (148594 flux in 732.67ms)

[dm3_scp.zip](https://github.com/keirf/greaseweazle/files/15053294/dm3_scp.zip)

**Reply by keirf** [reply]
Yes, that's a good dump. HxC finds it marginal, but `gw convert` is able to decode all sectors:
```
gw convert --format=ibm.scan dm3.scp dm3.img
```

You might get an even better dump by using the `--densel=l` option to `gw read`. But anyway, this shows the Greaseweazle is working okay, and the disk is okay. So the problems with SA400 are specific to Greaseweazle and that drive.

Edit: Oh, you did use `--dd=l` (which is the old equivalent to `--densel=l`) so ignore that bit but the rest stands.

**Comment by aurum79argentum47** [comment]
That is very good to know. I do appreciate all your help with this.

Not many people stand behind their products to that extent!

- Greg
